jconstraints-core/src/main/java/gov/nasa/jpf/constraints/smtlibUtility/smtsolver/SMTCMDContext.java

```diff
@@ -33,18 +34,28 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.PrintStream;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 public class SMTCMDContext extends SolverContext implements StoppableSolver {
   private Process p;
   SMTLibExportGenContext ctx;
 
   protected String[] command;
 
+  protected List<Expression<Boolean>> unsatCoreLastRun = null;
```

Comment:
The visibility in this class is odd. Some parts are private, some, package-private, and some protected? The visibilities (and class finality) should be checked.